engine/robot_state.py

```diff
@@ -0,0 +1,138 @@
+        max_velocity: the maximum velocity of the robot
+        radius: the radius of the robot
+    Preconditions:
+        x_pos is an int within the bounds of the user-defined traversal grid
+        y_pos is an int within the bounds of the user-defined traversal grid
+        heading is a float
+        epsilon is a float
+        max_velocity is a float
+        radius is a float
+    """
+    def __init__(self, x_pos, y_pos, heading, epsilon, max_velocity, radius):
+        """
+        Instance Attributes:
+            epsilon: dictates the turning radius of the robot. Lower epsilon results in tighter turning radius.
+            max_velocity: the maximum velocity of the robot
+            radius: the radius of the robot
+            width: width of the robot in cm
+            front_ultrasonic: the ultrasonic at the front of the robot, used for detecting obstacles
+            lf_ultrasonic: the ultrasonic at the front of the left side of the robot, used for boundary following
+            lb_ultrasonic: the ultrasonic at the back of the left side of the robot, used for boundary following
+            rf_ultrasonic: the ultrasonic at the front of the right side of the robot, used for boundary following
+            rb_ultrasonic: the ultrasonic at the back of the right side of the robot, used for boundary following
+            is_sim: False if the physical robot is being used, True otherwise
+            is_store: False if csv data should not be stored, True otherwise
+            position_kp: the proportional factor of the position PID
+            position_ki: the integral factor of the position PID
+            position_kd: the derivative factor of the position PID
+            position_noise: the flat amount of noise added to the robot's phase on each localization step
+            heading_kp: the proportional factor of the heading PID
+            heading_ki: the integral factor of the heading PID
+            heading_kd: the derivative factor of the heading PID
+            heading_noise: ?
+            phase: the phase of the robot
+            time_step: the amount of time that passes between each feedback loop cycle, should only be used if is_sim
+                is True
+            move_dist: the distance in meters that the robot moves per time dt
+            turn_angle: the angle in radians that the robot turns per time dt regardless of time step
+            plastic_weight: the weight of the trash the robot has collected
+            init_threshold (Double): Radius from initial position that will detect robot is back in initial position
+            goal_threshold (Double): Threshold from goal that will be detected as reaching goal in obstacle avoidance
+            noise_margin (Double): Margin from init_threshold that the robot has to leave before detecting robot has
+                left initial position
+        """
+        # User defined parameters
+        self.epsilon = epsilon
```

Comment:
The epsilon attribute is interesting to me... in the comment it says it's defining how tight the turning is - are we assuming in every single situation that we are turning, it will turn the same amount? If not, isn't it better to make this attribute dependent on other values? (Maybe not because the sand will cause a lot of slipping, but just some food for thought?)

Reply:
Same as above (robot_state + spec is what we had in robot's initialization).

Since epsilon seemed to have been defined a long time ago (maybe before I joined??), I'm not too sure how it should be defined. If we want to reevaluate our current attributes we can talk about it at subteam and make a separate PR for it (since this pr is more focused on the refactoring/moving the attributes from one place to another).